src/app/contribution-graph/contribution-graph.component.ts

```diff
@@ -134,4 +137,60 @@ export class ContributionGraphComponent implements OnInit {
     return activityDict;
   }
 
+
+  filterActivity(activity: any[]) {
+    const activityFiltered = [];
+    const startDate = new Date(this.startDate);
+    const endDate = this.getEndDate(startDate);
+    for (const [sDate, contNumber] of activity) {
+      const date = new Date(sDate);
+      if (date > startDate && date < endDate) {
+        activityFiltered.push([sDate, contNumber]);
+      }
+    }
+    return activityFiltered;
+  }
+  getTotalActivity(activity: any[]) {
+    // returns sum of all the contribution of the  *ONGOING* year
+    const activityFiltered = this.filterActivity(activity);
+    let totalActivity = 0;
+    for (const [key, contNumber] of activityFiltered) {
+      totalActivity += +contNumber;
+    }
+    return totalActivity;
+  }
+
+  getValue(activityCount: number) {
+    const totalActivityCount = this.getTotalActivity(this.activity);
+    if (totalActivityCount === 0 || activityCount === 0) {
+      return 0;
+    }
+    const ratio = activityCount / totalActivityCount * 100;
+    let value = 0;
+    if (ratio < 25) {
+      value = 1;
+    } else if (ratio < 50) {
+      value = 2;
+    } else if (ratio < 75) {
+      value = 3;
+    } else {
+      value = 4;
+    }
+    return value;
+  }
+
+
+  circleColor(i: number, j: number) {
+    const colorTable = this.colorScheme;
+    const id =  this.idCalc(i, j);
+    const date = this.id2date(id);
```

Comment:
I think line `185` and line `186` are used in several places in the code. Maybe it'd be good to create a function `getDate(col: number, row: number)` that directly returns the date?